frontend/amundsen_application/static/js/components/LineageGraph/wrapperLineage.ts

```diff
@@ -0,0 +1,290 @@
+    // .style('fill', (d) => (d._children ? 'lightsteelblue' : '#fff'));
+
+    // Add labels for the nodes
+    nodeEnter
+      .append('text')
+      .attr('dy', '.35em')
+      .attr('x', (d) => (d.y > 0 ? -15 : 15))
+      .attr('text-anchor', (d) => (d.y > 0 ? 'end' : 'start'))
+      .text(function (d, index) {
+        if (index === 0) return '';
+        // return `${d.data.data.schema}.${d.data.data.name}`;
+        return `${d.data.data.name}`;
+      });
+
+    nodeEnter
+      .append('text')
+      .attr('dx', '-.25em')
+      .attr('dy', '.25em')
+      .attr('class', 'plus')
+      .style('color', '#8b37ff')
+      .text((d) => (d._children ? '+' : ''));
+
+    // UPDATE
+    const nodeUpdate = nodeEnter.merge(node);
+
+    // Transition to the proper position for the node
+    nodeUpdate
+      .transition()
+      .duration(animationDuration)
+      .attr('transform', (d) => {
+        // DO NOT UPDATE THE POSITION OF ROOT NODE
+        if (d.parent === null) {
+          d.y = root.y0;
+          d.x = root.x0;
+        }
+        return 'translate(' + d.y + ',' + d.x + ')';
+      });
+
+    // Update the node attributes and style
+    nodeUpdate
+      .select('circle.node')
+      .attr('r', (d) => (d.depth === 0 ? 12 : 8))
+      .attr('cursor', 'pointer');
+
+    nodeUpdate.select('text.plus').text((d) => (d._children ? '+' : ''));
+
+    // Remove any exiting nodes
+    const nodeExit = node
+      .exit()
+      .transition()
+      .duration(animationDuration)
+      .attr('transform', () => `translate(${source.y},${source.x})`)
+      .remove();
+
+    // On exit reduce the node circles size to 0
+    nodeExit.select('circle').attr('r', 1e-6);
+
+    // On exit reduce the opacity of text labels
+    nodeExit.select('text').style('fill-opacity', 1e-6);
+
+    // ****************** links section ***************************
+
+    // Update the links...
+    const link = g.selectAll('path.link').data(links, function (d) {
+      return d.id;
+    });
+
+    // Enter any new links at the parent's previous position.
+    const linkEnter = link
+      .enter()
+      .insert('path', 'g')
+      .attr('class', 'link')
+      .attr('d', () => {
+        const o = { x: source.x0, y: source.y0 };
+        return generatePath(o, o);
+      });
+
+    // UPDATE
+    const linkUpdate = linkEnter.merge(link);
+
+    // Transition back to the parent element position
+    linkUpdate
+      .transition()
+      .duration(animationDuration)
+      .attr('d', (d) => generatePath(d, d.parent));
+
+    // Remove any exiting links
+    link
+      .exit()
+      .transition()
+      .duration(animationDuration)
+      .attr('d', () => {
+        const o = { x: source.x, y: source.y };
+        return generatePath(o, o);
+      })
+      .remove();
+
+    // Store the old positions for transition.bnm
+    nodes.forEach((d) => {
+      d.x0 = d.x;
+      d.y0 = d.y;
+    });
+  }
+};
```

Comment:
This looks OK, but I don't like the idea of having all the code in one function like this. This makes it really hard to test, extend, and modularize.

I would rather have us creating a graph component based on the Reusable API, which can be tested, configured and extended.

See an example here: https://github.com/britecharts/britecharts-react/blob/mi-typescript-move/src/charts/chartDemo.ts